tests/ctests/RotationTest.cpp

```diff
@@ -217,152 +217,126 @@ TEST(RotationTest, ToAxisAngle) {
-  EXPECT_NEAR(rotatedVY[4], 0.0, 1e-10);
-  EXPECT_NEAR(rotatedVY[5], 1.0, 1e-10);
-  ASSERT_EQ(rotatedVZ.size(), 6);
-  EXPECT_NEAR(rotatedVZ[0], 0.0, 1e-10);
-  EXPECT_NEAR(rotatedVZ[1], 0.0, 1e-10);
-  EXPECT_NEAR(rotatedVZ[2], 0.0, 1e-10);
-  EXPECT_NEAR(rotatedVZ[3], 1.0, 1e-10);
-  EXPECT_NEAR(rotatedVZ[4], 0.0, 1e-10);
-  EXPECT_NEAR(rotatedVZ[5], 0.0, 1e-10);
-}
-
-TEST(RotationTest, RotateWrongSizeAV) {
-  Rotation rotation;
-  ASSERT_NO_THROW(rotation({1.0, 1.0, 1.0}));
-  ASSERT_NO_THROW(rotation({1.0, 1.0, 1.0}, {1.0, 1.0}));
-}
-
-TEST(RotationTest, RotateWrongSizeVector) {
-  Rotation rotation;
-  ASSERT_THROW(rotation({1.0, 1.0, 1.0, 1.0}), std::invalid_argument);
+  State unitX({1.0, 0.0, 0.0, 0.0, 0.0, 0.0});
+  State unitY({0.0, 1.0, 0.0, 0.0, 0.0, 0.0});
+  State unitZ({0.0, 0.0, 1.0, 0.0, 0.0, 0.0});
+  State unitVX({0.0, 0.0, 0.0, 1.0, 0.0, 0.0});
+  State unitVY({0.0, 0.0, 0.0, 0.0, 1.0, 0.0});
+  State unitVZ({0.0, 0.0, 0.0, 0.0, 0.0, 1.0});
+  State rotatedX = rotation(unitX);
+  State rotatedY = rotation(unitY);
+  State rotatedZ = rotation(unitZ);
+  State rotatedVX = rotation(unitVX);
+  State rotatedVY = rotation(unitVY);
+  State rotatedVZ = rotation(unitVZ);
+  EXPECT_NEAR(rotatedX.position.x, 0.0, 1e-10);
+  EXPECT_NEAR(rotatedX.position.y, 1.0, 1e-10);
+  EXPECT_NEAR(rotatedX.position.z, 0.0, 1e-10);
+  EXPECT_NEAR(rotatedX.velocity.x, 0.0, 1e-10);
+  EXPECT_NEAR(rotatedX.velocity.y, 0.0, 1e-10);
+  EXPECT_NEAR(rotatedX.velocity.z, 0.0, 1e-10);
+  EXPECT_NEAR(rotatedY.position.x, 0.0, 1e-10);
+  EXPECT_NEAR(rotatedY.position.y, 0.0, 1e-10);
+  EXPECT_NEAR(rotatedY.position.z, 1.0, 1e-10);
+  EXPECT_NEAR(rotatedY.velocity.x, 0.0, 1e-10);
+  EXPECT_NEAR(rotatedY.velocity.y, 0.0, 1e-10);
+  EXPECT_NEAR(rotatedY.velocity.z, 0.0, 1e-10);
+  EXPECT_NEAR(rotatedZ.position.x, 1.0, 1e-10);
+  EXPECT_NEAR(rotatedZ.position.y, 0.0, 1e-10);
+  EXPECT_NEAR(rotatedZ.position.z, 0.0, 1e-10);
+  EXPECT_NEAR(rotatedZ.velocity.x, 0.0, 1e-10);
+  EXPECT_NEAR(rotatedZ.velocity.y, 0.0, 1e-10);
+  EXPECT_NEAR(rotatedZ.velocity.z, 0.0, 1e-10);
+  EXPECT_NEAR(rotatedVX.position.x, 0.0, 1e-10);
+  EXPECT_NEAR(rotatedVX.position.y, 0.0, 1e-10);
+  EXPECT_NEAR(rotatedVX.position.z, 0.0, 1e-10);
+  EXPECT_NEAR(rotatedVX.velocity.x, 0.0, 1e-10);
+  EXPECT_NEAR(rotatedVX.velocity.y, 1.0, 1e-10);
+  EXPECT_NEAR(rotatedVX.velocity.z, 0.0, 1e-10);
+  EXPECT_NEAR(rotatedVY.position.x, 0.0, 1e-10);
+  EXPECT_NEAR(rotatedVY.position.y, 0.0, 1e-10);
+  EXPECT_NEAR(rotatedVY.position.z, 0.0, 1e-10);
+  EXPECT_NEAR(rotatedVY.velocity.x, 0.0, 1e-10);
+  EXPECT_NEAR(rotatedVY.velocity.y, 0.0, 1e-10);
+  EXPECT_NEAR(rotatedVY.velocity.z, 1.0, 1e-10);
+  EXPECT_NEAR(rotatedVZ.position.x, 0.0, 1e-10);
+  EXPECT_NEAR(rotatedVZ.position.y, 0.0, 1e-10);
+  EXPECT_NEAR(rotatedVZ.position.z, 0.0, 1e-10);
+  EXPECT_NEAR(rotatedVZ.velocity.x, 1.0, 1e-10);
+  EXPECT_NEAR(rotatedVZ.velocity.y, 0.0, 1e-10);
+  EXPECT_NEAR(rotatedVZ.velocity.z, 0.0, 1e-10);
```

Comment:
We will probably want some sort of vec3d comparison tool for testing. Not right now, but eventually.

Reply:
I know the Vec3d object is just a struct but could we still overload the equivalence operator?